frontend/public/components/catalog/catalog-items.jsx

```diff
@@ -12,101 +12,147 @@ import {Modal} from 'patternfly-react/dist/js/components/Modal';
     this.closeOverlay = this.closeOverlay.bind(this);
   }
 
-  componentDidUpdate(prevProps, prevState) {
-    const { filters, filterCounts, activeTabs, categories } = this.state;
+  componentDidMount () {
+    this.setState(this.getUpdatedStateFromURLParams(window.location.search));
+  }
+
+  componentDidUpdate(prevProps) {
+    const { filters, selectedCategory } = this.state;
     const { items } = this.props;
 
     if (items !== prevProps.items) {
-      const newCategories = categorizeItems(items);
-      this.setState(this.getCategoryState(activeTabs, newCategories));
-      if (this.hasActiveFilters(filters)) {
-        const filteredItems = this.filterItems(items);
-        const filteredCategories = recategorizeItems(filteredItems, newCategories);
-        this.setState(this.getCategoryState(activeTabs, filteredCategories));
-      }
-      this.setState(this.getFilterCounts(activeTabs, filters, filterCounts, newCategories));
+      const categories = categorizeItems(items);
+      this.setState(this.getUpdatedState(categories, selectedCategory, filters));
     }
+  }
 
-    if (filters !== prevState.filters) {
-      const filteredItems = this.filterItems(items);
-      const newCategories = recategorizeItems(filteredItems, categories);
-      this.setState(this.getCategoryState(activeTabs, newCategories));
-    }
+  getUpdatedState (categories, selectedCategory, filters) {
+    const { items } = this.props;
+    const { filterCounts } = this.state || {};
 
-    if (activeTabs !== prevState.activeTabs) {
-      this.setState(this.getCategoryState(activeTabs, categories));
+    if (!items) {
+      return;
     }
 
-    // filter counts are updated when new Category tab is selected or filter by name changed
-    if (activeTabs !== prevState.activeTabs || prevState.filters.byKeyword !== filters.byKeyword ) {
-      this.setState(this.getFilterCounts(activeTabs, filters, filterCounts, categories));
-    }
+    const updateFilterCounts = filterCounts ||
+      {
+        byType: {
+          clusterServiceClasses: 0,
+          imageStreams: 0,
+        },
+      };
+
+    const filteredItems = this.filterItems(items, filters);
+
+    const newCategories = recategorizeItems(filteredItems, categories);
+
+    return {
+      filters,
+      ...CatalogTileViewPage.getCategoryState(selectedCategory, newCategories),
+      filterCounts: this.getFilterCounts(selectedCategory, filters, updateFilterCounts, newCategories)
+    };
+  }
+
+  getUpdatedStateFromURLParams(paramsString) {
+    const { categories } = this.state;
+    const searchParams = new URLSearchParams(paramsString);
+    const categoryParam = searchParams.get(CATEGORY_URL_PARAM);
+    const keywordParam = searchParams.get(KEYWORD_URL_PARAM);
+    const byTypeParam = searchParams.get(TYPE_URL_PARAM);
+    const typeFilters = byTypeParam ? JSON.parse(byTypeParam) : [];
+    const curCategory = categoryParam ? JSON.parse(categoryParam) : ['all'];
```

Comment:
We should catch exceptions everywhere we call JSON parse. Otherwise a bad value would whitescreen the console.

You might find `_.attempt` useful.

https://lodash.com/docs/4.17.11#attempt

Reply:
added try/catch